src/os_cloud_nodelet.cpp

```diff
@@ -33,6 +34,30 @@ using ouster_ros::PacketMsg;
 using sensor::UDPProfileLidar;
 using namespace std::chrono_literals;
 
+namespace {
+template <typename T, typename UnaryPredicate>
+int find_if_reverse(const Eigen::Array<T, -1, 1>& array,
+                    UnaryPredicate predicate) {
+    auto p = array.data() + array.size() - 1;
+    do {
+        if (predicate(*p)) return p - array.data();
+    } while (p-- != array.data());
+    return -1;
+}
```

Comment:
Initially had it like this but found it that does not cross compile.

Reply:
Sorry I think that you meant something else

Reply:
I wouldn't want to create a `reverse` of the array just so I could find the last element. while you could easily find the item by iterating the array backward in-place..

Reply:
Originally I thought you were suggesting me to use `std::reverse_iterator`, this is a more appropriate solution and you can past it to std::find_if and it would traverse the array in-place backwards. The only issue it wouldn't cross-compile (failed on gcc-9).